From: ryanking13 <def6488@gmail.com>
Date: Thu, 13 Aug 2026 18:30:00 +0900
Subject: [PATCH] Use Emscripten endian functions

Emscripten provides the standard endian conversion functions in endian.h,
but it was not detected properly.

TODO: upstream this patch to psycopg

---
 psycopg_c/_psycopg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/psycopg_c/_psycopg.c b/psycopg_c/_psycopg.c
--- a/psycopg_c/_psycopg.c
+++ b/psycopg_c/_psycopg.c
@@ -1240,7 +1240,7 @@
 
 #endif
 
-#if defined(__linux__) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__CYGWIN__) || defined(__EMSCRIPTEN__)
 
 #   include <endian.h>
 
